### PR TITLE
lua: Add automatic return and = handling in console.

### PR DESCRIPTION
**Pull request by mniip**

### CODE REVIEW

File: plugins/lua/lua.c [hunk 1]
```diff
@@ -1481,15 +1481,30 @@ static void inject_string(script_info *info, char const *line)
 {
 	lua_State *L = info->state;
 	int base, top;
+	char *program, *ret_program;
+
+	if(line[0] == '=')
+		program = g_strconcat("return ", line + 1, NULL);
+	else
+		program = g_strdup(line);
+	ret_program = g_strconcat("return ", program, NULL);
```
Comment (Arnavion) [inline plugins/lua/lua.c]: Do you really want to do `return return` or do you just want to do

```
1. if line[0] == '=' then line = line + 1
2. program = "return " + line
```

Comment (Arnavion) [inline plugins/lua/lua.c]: Also how does this interact with multi-line lua statements. Do you need to wrap `line` in something like an IIFE and return that instead?

**Comment by mniip** [comment]
`return return` is not an issue because that is obviously a syntactic error and the first loadbuffer will fail and the other one will be used.

**Comment by Arnavion** [comment]
I'm trying to say that rather than having the possibilty of constructing a `return return` that's guaranteed to not be what you want, you could just strip the leading `=`

**Comment by mniip** [comment]
Would that make much difference? Seems like it would only complicate the logic by a bunch.

**Comment by Arnavion** [comment]
Umm...

```c
if(line[0] == '=')
	program = g_strconcat("return ", line + 1, NULL);
else
	program = g_strdup(line);
ret_program = g_strconcat("return ", program, NULL);
```

becomes

```c
if(line[0] == '=')
	line = line + 1;
ret_program = g_strconcat("return ", line, NULL);
```

**Comment by mniip** [comment]
Hmm I suppose.

**Comment by Arnavion** [comment]
This still breaks multiple lines separated by `;`, right?

**Comment by mniip** [comment]
Not at all. `return x(); y()` is a syntactic error.

**Pull request merged**